src/OrchardCore.Modules/OrchardCore.Users/Services/SafeBoolModelBinder.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Users.ViewModels;
+
+namespace OrchardCore.Users.Services;
+
+/// <summary>
+/// Model binder to produce a validation error when a Boolean field contains a value that's not a valid bool. The
+/// default model binder would throw a <see cref="FormatException"/>. That's an issue for <see
+/// cref="LoginViewModel.RememberMe"/> that can get invalid values from automated cracking attempts.
+/// </summary>
+internal class SafeBoolModelBinder : IModelBinder
+{
+    public Task BindModelAsync(ModelBindingContext bindingContext)
+    {
+        ArgumentNullException.ThrowIfNull(nameof(bindingContext));
+
+        var valueProviderResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName);
+
+        if (valueProviderResult == ValueProviderResult.None)
+        {
+            return Task.CompletedTask;
+        }
+
```

Comment:
@Piedone did you actually try this PR out? I don't think it would fix your problem just yet. I think you are missing something like this here

```
        if (valueProviderResult.FirstValue == "1")
        {
            bindingContext.Result = ModelBindingResult.Success(true);
            return Task.CompletedTask;
        }

        if (valueProviderResult.FirstValue == "0")
        {
            bindingContext.Result = ModelBindingResult.Success(false);
            return Task.CompletedTask;
        }
```

Reply:
Yes, I did try it out of course, and such a behavior is not what I intended. This is not about supporting any other boolean-like values that no well-meaning client would send. Rather, this is about preventing a 5xxx error when a non-boolean value is sent, and causing a validation error instead.

Reply:
Even better. Awesome! Did the exception go away? You may want try a binder with `bool?` as well to make sure that still works with no exception as expected

Reply:
Yes, there's no exception now.

`bool?` can't be used with `asp-for` so the issue doesn't arise there (and otherwise the default model binder handles it correctly, so a gibberish value will result in `false`, not an exception). I'm not aware of any other case other than the tag helper where a `FormatException` would be thrown.